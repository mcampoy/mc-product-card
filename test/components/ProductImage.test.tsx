import * as React from 'react';
import renderer from 'react-test-renderer'
import ProductCard, { ProductImage } from '../../src/components';
import { product2 } from '../data/product';

describe('ProdctImage', () => {
  test('debe mostrar el componente correctamente con la imagen personalizada', () => {

    const wrapper = renderer.create(
      <ProductImage img="https://hola.jpg" />
    )

    expect( wrapper.toJSON() ).toMatchSnapshot();

  });

  test('debe mostrar el componente con la imagen del producto', () => {
    const wrapper = renderer.create(
      <ProductCard product={ product2 }>
        {
          () => (
            <ProductImage img={product2.img} />
          )
        }
      </ProductCard>
    )

    expect( wrapper.toJSON() ).toMatchSnapshot();

  })
});
