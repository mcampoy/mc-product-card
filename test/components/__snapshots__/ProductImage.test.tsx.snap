// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProdctImage debe mostrar el componente con la imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="undefined productImg"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProdctImage debe mostrar el componente correctamente con la imagen personalizada 1`] = `
<img
  alt="Product"
  className="undefined productImg"
  src="https://hola.jpg"
/>
`;
